lib/spack/spack/cmd/env.py

```diff
@@ -118,19 +209,40 @@ def env_activate_setup_parser(subparser):
         help="decorate the command line prompt when activating",
     )
 
-    env_options = subparser.add_mutually_exclusive_group()
-    env_options.add_argument(
+    subparser.add_argument(
         "--temp",
         action="store_true",
         default=False,
         help="create and activate an environment in a temporary directory",
     )
-    env_options.add_argument(
-        "-d", "--dir", default=None, help="activate the environment in this directory"
+    subparser.add_argument(
+        "--create",
+        action="store_true",
+        default=False,
+        help="if the environment doesn't exist, create it before activating",
+    )
+    subparser.add_argument(
+        "--envfile",
```

Comment:
This changes the user interface as it relates to providing an environment file so, if this approach is preferred, the documentation has to change (e.g., https://spack.readthedocs.io/en/latest/environments.html#creating-a-managed-environment) and the tutorial has to change (https://spack-tutorial.readthedocs.io/en/latest/tutorial_environments.html#using-spack-yaml).

Reply:
All these commands in the docs and tutorial are still valid.  We are just adding another, abbreviated, way to do the same thing.  I can add some documentation to the creating a managed environment section, but I think we should leave the tutorial as is for now.  I would suggest when I get `spack env create --activate` added in the next PR we could add that to the tutorial.  I think that goes more naturally with the flow there.

Reply:
Actually looking at the flow in the docs I think adding to the creating a managed env section is not worth doing. The activate section with the new docs follows directly after it and explains what the `--create` flag will do. I updated the documentation a little more so hopefully that is sufficient?